src/main/haskell/language-kore/src/Data/Kore/AST/Builders.hs

```diff
@@ -371,3 +399,69 @@ next_ =
                 , nextChild  = pattern1
                 }
         )
+
+{-|'equalsSortParam' is the sort param implicitly used for 'equals' when no
+other sort can be inferred. This parameter is assumed not to be used in
+any pattern, except for top-level pattern of an axiom. Using it will have
+unpredictable effects.
+-}
+equalsSortParam :: MetaOrObject level => proxy level -> SortVariable level
```

Comment:
I don't think it's a food idea to move this out of the implicit Kore implementation because it actually is an implementation detail. Having it here exposes more people to the dangers of using it in wrongly. How about moving it back and changing the *equalsAxiom_ builders to take an extra argument, and redefining the *equalsAxiom functions in the Kore implementation to pass the equalsSortParam to the underscore versions? You can do a similar thing for tests.